vyvolavac/Vyvolavac.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/python
+"""
+    Vyvolavac.py: Randomly selects a student with a certain probability.
+    1st argument input_path - input path of student names with (or without) statistics, input file should contain
+    one student per line and the statistics are tab delimited
+    2nd argument (optional): output_path  - output path for new statistics  
+"""
+
+__maintainer__ = "Dagy Tran"
+__version__ = "0.1 beta"
+__status__ = "Development"
+
+
+import sys
+from random import randint
+import matplotlib.pyplot as plt
+
+
+def load_statistics(input_path):
+    stats = {}
+    should_reset = False
+    with open(input_path, 'r') as input_file:
+        for line in input_file:
+            split_line = line.split('\t')
+
+            if len(split_line) == 1:
+                stats[split_line[0].strip()] = (0, 0)
+                should_reset = True
+            elif len(split_line) == 3:
+                stats[split_line[0]] = (int(split_line[1]), int(split_line[2]))
+
+    if should_reset:
+        stats = reset_stats(stats, 2)
+
+    return stats
+
+
+def reset_stats(stats_dict, multiplier=2):
+    number_of_students = len(stats_dict)
+
+    for key, value in stats_dict.items():
+        stats_dict[key] = (0, multiplier * number_of_students)
+    return stats_dict
+
+
+def save_statistics(output_path, stats_dictionary):
+    with open(output_path, 'w') as output_file:
+        lines = list(map(lambda kv: '{}\t{}\t{}'.format(kv[0], kv[1][0], kv[1][1]), stats_dictionary.items()))
+        output_file.write('\n'.join(lines))
+
+
+def redistribute(stats_dictionary, key):
+    new_stats = {}
+    number_of_students = len(stats_dictionary)
+    if key not in stats_dictionary:
+        print('Error: User not in dictionary.')
+        return stats_dictionary
+
+    student_points = stats_dictionary[key][1]
+    if student_points < number_of_students - 1:
+        print('Warning: User {} has not enough points to redistribute. Points {}. Returning unchanged stats'.format(key,
```

Comment:
Pokud nastane tato situace, tak mi to potom vypíše na výstup již vybraného studenta, ale statistiky zůstanou stejné. Očekával bych spíš, že se znovu zavolá funkce ``select_student``